master/internal/provisioner/aws_spot.go

```diff
@@ -186,11 +195,16 @@ func (c *awsCluster) listSpot(ctx *actor.Context) ([]*Instance, error) {
 	}
 
 	if canceledButInstanceRunningReqs.numReqs() > 0 {
-		ctx.Log().Infof(
+		ctx.Log().Debugf(
 			"terminating EC2 instances associated with canceled spot requests: %s",
 			strings.Join(canceledButInstanceRunningReqs.idsAsList(), ","),
 		)
-		c.terminateOnDemand(ctx, canceledButInstanceRunningReqs.idsAsListOfPointers())
+		_, err = c.terminateInstances(canceledButInstanceRunningReqs.instanceIds())
+		if err != nil {
+			ctx.Log().
+				WithError(err).
+				Debugf("cannot terminate EC2 instances associated with canceled spot requests")
```

Comment:
question: why is it a debug log? Is it because it's spamming?

Reply:
Yeah. It could be info. It's not as bad as the others logs.